spring-integration-kafka/src/main/java/org/springframework/integration/kafka/outbound/KafkaProducerMessageHandler.java

```diff
@@ -257,13 +282,14 @@ public void setSendTimeout(long sendTimeout) {
 
 	/**
 	 * Specify a SpEL expression to evaluate a timeout in milliseconds for how long this
-	 * {@link KafkaProducerMessageHandler} should wait wait for send operation
-	 * results. Defaults to 10 seconds. The timeout is applied only in {@link #sync} mode.
+	 * {@link KafkaProducerMessageHandler} should wait wait for send operation results.
+	 * Defaults to the kafka {@code delivery.timeout.ms} property + 5 seconds. The timeout
+	 * is applied only in {@link #sync} mode.
 	 * @param sendTimeoutExpression the {@link Expression} for timeout to wait for result
-	 * fo send operation.
+	 * for a send operation.
 	 * @since 2.1.1
 	 */
-	public void setSendTimeoutExpression(Expression sendTimeoutExpression) {
+	public final void setSendTimeoutExpression(Expression sendTimeoutExpression) {
```

Comment:
Looks like you have changed your mind and decided to not use this setter in the ctor.
I mean doesn't look like there is still a justification for a `final`...